crates/router/src/core/api_keys.rs

```diff
@@ -0,0 +1,220 @@
+use common_utils::{date_time, errors::CustomResult, fp_utils};
+use error_stack::{report, IntoReport, ResultExt};
+use masking::{PeekInterface, Secret};
+use router_env::{instrument, tracing};
+
+use crate::{
+    consts,
+    core::errors::{self, RouterResponse, StorageErrorExt},
+    db::StorageInterface,
+    services::ApplicationResponse,
+    types::{api, storage, transformers::ForeignInto},
+    utils,
+};
+
+// Defining new types `PlaintextApiKey` and `HashedApiKey` in the hopes of reducing the possibility
+// of plaintext API key being stored in the data store.
+pub struct PlaintextApiKey(Secret<String>);
+pub struct HashedApiKey(String);
+
+impl PlaintextApiKey {
+    const HASH_KEY_LEN: usize = 32;
+
+    const PREFIX_LEN: usize = 12;
+
+    pub fn new(length: usize) -> Self {
+        let env = match router_env::env::which() {
+            router_env::Env::Development => "dev",
+            router_env::Env::Sandbox => "snd",
+            router_env::Env::Production => "prd",
```

Comment:
Hey, can we define this somewhere in the `router_env` itself, we have a similar match case in `admin.rs`. have it in a single place will help in the future, if a need of performing a similar match arises 

https://github.com/juspay/hyperswitch/blob/3292960996834f93095f96f05b18ce28adceb374/crates/router/src/core/admin.rs#L23-L27